docs: improve CCM gas budget and CCM estimations docs (#571)
chainflip-io/chainflip-docs: pages/swapping/integrations/advanced/cross-chain-messaging/evm-ccm.mdx

- The gas budget provided is not enough to execute the the receiver's logic on the destination chain.
- The receiver contract not implementing the correct interface (or being an EOA).
- The logic executed on the receiver's contract reverts.

Chainflip will use TSS to sign over a valid payload for the destination chain and try to broadcast it. If a transaction can't be broadcasted for any reason or it reverts when running a simulation, the signed payload will be publicly accessible on the Chainflip State Chain for the user. The user can then sign over the Chainflip-signed payload and broadcast the transaction.

This is particularly useful if the transaction is reverting due to not having enough gas to execute the user logic. In that scenario the user can sign that payload and broadcast it with a higher amount of gas. 

There is nothing Chainflip can do if the transaction is reverting due to the receiver contract not implementing the correct interface or it's logic reverting. As described [above](#ccm-receiver-smart-contract-interface) it is highly recommended to use try/catch statments to handle reverts in the receiver contract.

> The Chainflip-signed payload will be valid until two key rotations have taken place. On mainnet this is currently up to 6 days.

Reviewer: ```suggestion
The Chainflip-signed payload will be valid until two key rotations have taken place. On mainnet this is currently **up to 6 days**.
```

> The Chainflip-signed payload will be valid until two key rotations have taken place. On mainnet this is currently up to 6 days.

Reviewer 2: think the minimum could be important here
```suggestion
The Chainflip-signed payload will be valid until two key rotations have taken place. On mainnet this is currently at least 3 days from the time of signing.
```

<Callout type="warning">
The signed payload can be broadcasted with different gas settings but it can't modify any value in that payload (data) such as the message. Therefore, always make sure to test the receiving logic and ensure it won't revert before initiating a swap.

> It is strongly encouraged to use try/catch statments to handle reverts in the receiver contract.

Reviewer: ```suggestion
**We strongly recommend using try/catch statements to handle reverts in the receiver contract.**.
```
